content/browser/secure_embed_connector_impl.cc

```diff
@@ -104,42 +150,28 @@ SecureEmbedConnector::Delegate* SecureEmbedConnectorImpl::GetDelegate() {
   return delegate_;
 }
 
-void SecureEmbedConnectorImpl::ForwardKeyboardEvent(
-    const blink::WebKeyboardEvent& keyboard_event) {
-  if (!guest_web_contents_ || !view_) {
-    return;
-  }
-
-  input::NativeWebKeyboardEvent native_event(
-      keyboard_event, GetParentRenderWidgetHostView()->GetNativeView());
-
-  RenderWidgetHostImpl* target_host = view_->host();
-
-  // If there are multiple widgets on the page (such as when there are
-  // out-of-process iframes), pick the one that should process this event.
-  if (target_host->delegate()) {
-    target_host =
-        target_host->delegate()->GetFocusedRenderWidgetHost(target_host);
-  }
-  if (!target_host) {
-    return;
-  }
-
-  target_host->ForwardKeyboardEvent(native_event);
-}
-
 void SecureEmbedConnectorImpl::SetFocus(bool focused,
                                         blink::mojom::FocusType focus_type) {
   if (!guest_web_contents_ || !view_) {
     return;
   }
 
   view_->host()->SetPageFocus(focused);
-  if (focused && (focus_type == blink::mojom::FocusType::kForward ||
-                  focus_type == blink::mojom::FocusType::kBackward)) {
-    static_cast<RenderViewHostImpl*>(guest_web_contents_->GetRenderViewHost())
-        ->SetInitialFocus(
-            /*reverse=*/focus_type == blink::mojom::FocusType::kBackward);
+  if (focused) {
+    if (focus_type == blink::mojom::FocusType::kForward ||
+        focus_type == blink::mojom::FocusType::kBackward) {
+      static_cast<RenderViewHostImpl*>(guest_web_contents_->GetRenderViewHost())
+          ->SetInitialFocus(
+              /*reverse=*/focus_type == blink::mojom::FocusType::kBackward);
+    }
+    // Ensure that the embedded frame tree is the focused frame tree if it is
+    // not already the focused frame tree when the plugin becomes in focus.
+    // kPage doesn't involved focused frame change, skip the check if it is
+    // kPage.
+    if ((focus_type != blink::mojom::FocusType::kPage) &&
+        !guest_web_contents_->ContainsOrIsFocusedWebContents()) {
```

Comment:
To simplify the code, can we ignore `ContainsOrIsFocusedWebContents` and always call  `SetFocusedFrameTree()`?